spec/features/consumer/shopping/cart_spec.rb

```diff
@@ -73,15 +74,15 @@
 
       it "shows the quantities saved, not those submitted" do
         # Given we load the page with 3 on hand, then the number available drops to 2
+        variant.update_attributes! on_hand: 3
         visit spree.cart_path
         variant.update_attributes! on_hand: 2
 
         fill_in "order_line_items_attributes_0_quantity", with: '4'
-
         click_button 'Update'
 
-        page.should have_field "order[line_items_attributes][0][quantity]", with: '1'
-        page.should have_content "Insufficient stock available, only 2 remaining"
+        expect(page).to have_content "Insufficient stock available, only 2 remaining"
```

Comment:
It's always nice to do re-factoring and code changes in separate commits. I got confused by the re-ordering of these two lines.